app/src/main/java/com/keylesspalace/tusky/adapter/ReportNotificationViewHolder.kt

```diff
@@ -0,0 +1,96 @@
+ *
+ * This program is free software; you can redistribute it and/or modify it under the terms of the
+ * GNU General Public License as published by the Free Software Foundation; either version 3 of the
+ * License, or (at your option) any later version.
+ *
+ * Tusky is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY; without even
+ * the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU General
+ * Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License along with Tusky; if not,
+ * see <http://www.gnu.org/licenses>. */
+
+package com.keylesspalace.tusky.adapter
+
+import android.content.Context
+import android.graphics.PorterDuff
+import android.graphics.drawable.Drawable
+import androidx.annotation.ColorRes
+import androidx.annotation.DrawableRes
+import androidx.core.content.ContextCompat
+import androidx.recyclerview.widget.RecyclerView
+import at.connyduck.sparkbutton.helpers.Utils
+import com.keylesspalace.tusky.R
+import com.keylesspalace.tusky.adapter.NotificationsAdapter.NotificationActionListener
+import com.keylesspalace.tusky.databinding.ItemReportNotificationBinding
+import com.keylesspalace.tusky.entity.Report
+import com.keylesspalace.tusky.entity.TimelineAccount
+import com.keylesspalace.tusky.util.TimestampUtils
+import com.keylesspalace.tusky.util.emojify
+import com.keylesspalace.tusky.util.loadAvatar
+import com.keylesspalace.tusky.util.unicodeWrap
+import java.util.Date
+
+class ReportNotificationViewHolder(
+    private val binding: ItemReportNotificationBinding,
+) : RecyclerView.ViewHolder(binding.root) {
+
+    fun setupWithReport(reporter: TimelineAccount, report: Report, animateAvatar: Boolean, animateEmojis: Boolean) {
+        val reporterName = reporter.name.unicodeWrap().emojify(reporter.emojis, itemView, animateEmojis)
+        val reporteeName = report.targetAccount.name.unicodeWrap().emojify(report.targetAccount.emojis, itemView, animateEmojis)
+        val icon = ContextCompat.getDrawable(itemView.context, R.drawable.ic_flag_24dp)?.apply {
+            setColorFilter(itemView.context.getColor(R.color.tusky_blue), PorterDuff.Mode.SRC_ATOP)
+        }
+        binding.notificationTopText.setCompoundDrawablesWithIntrinsicBounds(icon, null, null, null)
+        binding.notificationTopText.text = itemView.context.getString(R.string.notification_header_report_format, reporterName, reporteeName)
+        binding.notificationSummary.text = itemView.context.getString(R.string.notification_summary_report_format, TimestampUtils.getRelativeTimeSpanString(itemView.context, report.createdAt.time, Date().time), report.status_ids?.size ?: 0)
+        binding.notificationCategory.text = report.category
+
+        // Fancy avatar inset
+        val padding = Utils.dpToPx(binding.notificationReporteeAvatar.context, 12)
+        binding.notificationReporteeAvatar.setPaddingRelative(0, 0, padding, padding)
```

Comment:
Can't this be in the layout xml?

Reply:
I tried, but I was never able to get it to work?
The other notification and status views are doing it this way as well…

Reply:
ok, let's leave it then